Adding Hashicorp vault support (#655)
mozilla/sops: README.rst

	$ vault status
	Key             Value
	---             -----
	Seal Type       shamir
	Initialized     true
	Sealed          false
	Total Shares    1
	Threshold       1
	Version         1.2.0
	Cluster Name    vault-cluster-618cc902
	Cluster ID      e532e461-e8f0-1352-8a41-fc7c11096908
	HA Enabled      false

	$ # It is required to enable a transit engine if not already done (It is suggested to create a transit engine specifically for sops, in which it is possible to have multiple keys with various permission levels)
	$ vault secrets enable -path=sops transit
	Success! Enabled the transit secrets engine at: sops/

	$ # Then create one or more keys
	$ vault write sops/keys/firstkey type=rsa-4096
	Success! Data written to: sops/keys/firstkey

	$ vault write sops/keys/secondkey type=rsa-2048
	Success! Data written to: sops/keys/secondkey

	$ vault write sops/keys/thirdkey type=chacha20-poly1305
	Success! Data written to: sops/keys/thirdkey

	$ sops --hc-vault-transit $VAULT_ADDR/v1/sops/keys/firstkey vault_example.yml

	$ cat <<EOF > .sops.yaml
	creation_rules:
		- path_regex: \.dev\.yaml$
		  hc_vault_transit_uri: "$VAULT_ADDR/v1/sops/keys/secondkey"

> 		- path_regex: \.prod\.yaml$

Reviewer: Was this regular expression supposed to match the file paths mentioned on line 348 below?

	 $ sops --decrypt test.enc.yaml

Encrypting using Hashicorp Vault
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

> To understand how to deploy Vault securely this is not the place, we assume you have a instance (or more) of Vault running and you have privileged access to it. 

Reviewer 2: ```suggestion
We assume you have an instance (or more) of Vault running and you have privileged access to it. For instructions on how to deploy a secure instance of Vault, refer to Hashicorp's official documentation.
```

To easily deploy Vault locally: (DO NOT DO THIS FOR PRODUCTION!!!) 

.. code:: bash

	$ docker run -d -p8200:8200 vault:1.2.0 server -dev -dev-root-token-id=toor

.. code:: bash

> 	$ # Subsitute this with the address of the vault running

Reviewer 2: ```suggestion
	$ # Substitute this with the address Vault is running on
```

	$ export VAULT_ADDR=http://127.0.0.1:8200 

	$ # this may not be necessary in case you previously used `vault login` for production use
	$ export VAULT_TOKEN=toor 
	

> 	$ # to check if Vault started correctly and configured

Reviewer 2: ```suggestion
	$ # to check if Vault started and is configured correctly
```

	$ vault status
	Key             Value
	---             -----
	Seal Type       shamir
	Initialized     true
	Sealed          false
	Total Shares    1
	Threshold       1
	Version         1.2.0
	Cluster Name    vault-cluster-618cc902
	Cluster ID      e532e461-e8f0-1352-8a41-fc7c11096908
	HA Enabled      false

	$ # It is required to enable a transit engine if not already done (It is suggested to create a transit engine specifically for sops, in which it is possible to have multiple keys with various permission levels)
	$ vault secrets enable -path=sops transit

> 	Success! Enabled thˇe transit secrets engine at: sops/

Reviewer 2: ```suggestion
	Success! Enabled the transit secrets engine at: sops/
```

	$ # Then create one or more keys
	$ vault write sops/keys/firstkey type=rsa-4096
	Success! Data written to: sops/keys/firstkey

	$ vault write sops/keys/secondkey type=rsa-2048
	Success! Data written to: sops/keys/secondkey

	$ vault write sops/keys/thirdkey type=chacha20-poly1305
	Success! Data written to: sops/keys/thirdkey

	$ sops --hc-vault-transit $VAULT_ADDR/v1/sops/keys/firstkey vault_example.yml

> 	$ cat <<EOF >.sops.yaml

Reviewer 2: ```suggestion
	$ cat <<EOF > .sops.yaml
```

	creation_rules:
		- path_regex: \.dev\.yaml$
		  hc_vault_transit_uri: "$VAULT_ADDR/v1/sops/keys/secondkey"

> 		- \.prod\.yaml$

Reviewer 2: ```suggestion
		- path_regex: \.prod\.yaml$
```
